### PR TITLE
fix: avoid flattening single-item-list vertex values if the attribute…

### PR DESCRIPTION
**Pull request by kadoshms**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

1. Fixed vertex attribute values serializiation on graph creation  - verteices with single-item values would be flattned, for instance:
```
 cidr_blocks: ["0.0.0.0/0"]
```

becomes:
```
cidr_blocks: "0.0.0.0/0"
```

On graph creations, fixed this to only apply to attributes without a '.' notation.

2. Added two attribute solvers `length_equals` and `length_not_equals`

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/checks_infra/solvers/attribute_solvers/length_equals_attribute_solver.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        if vertex.get(attribute) is None:  # type:ignore[arg-type]  # due to attribute can be None
+            return False
+
+        attr = vertex.get(attribute)  # type:ignore[arg-type]  # due to attribute can be None
+        # if this value contains an underendered variable, then we cannot evaluate the check,
+        # so return True (since we cannot return UNKNOWN)
+        if self._is_variable_dependant(attr, vertex['source_']):
+            return True
+
+        if isinstance(attr, Sized):
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/length_equals_attribute_solver.py:24]: nice 👍 😄

File: checkov/common/checks_infra/solvers/attribute_solvers/length_equals_attribute_solver.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            return len(attr) == force_integer(self.value)
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/length_equals_attribute_solver.py:25]:
```suggestion
            return len(attr) == force_int(self.value)
```
don't need to add a new function, it already exists in the same file 😄

File: checkov/common/util/type_forcers.py [hunk 3]
```diff
@@ -44,6 +44,15 @@ def force_float(var: Any) -> float | None:
         return None
 
 
+def force_integer(var: Any) -> int | None:
```
Comment (gruebel) [inline checkov/common/util/type_forcers.py:47]: you can remove it and just use `force_int` 😉

**Review by gruebel — COMMENTED** [review]
nice job 🍻 looking forward to the next operators 😄

File: checkov/common/checks_infra/solvers/attribute_solvers/length_equals_attribute_solver.py (hunk 2 above)
Comment (kadoshms) [reply checkov/common/checks_infra/solvers/attribute_solvers/length_equals_attribute_solver.py:25]: missed that ^^

**Review by nimrodkor — CHANGES_REQUESTED** [review]
@kadoshms Regarding this:
> verteices with single-item values would be flattned

Could be very problematic. Let's say I want to query the attribute `cidr_blocks`. Should I use string operators on it, or list operators? The name of the field is in plural, so I expect a list... Why is this change needed?

**Comment by gruebel** [comment]
@nimrodkor 
> Could be very problematic. Let's say I want to query the attribute cidr_blocks. Should I use string operators on it, or list operators? The name of the field is in plural, so I expect a list... Why is this change needed?

This PR is fixing the behaviour of flattening single value lists 😄

**Review by nimrodkor — APPROVED** [review]
Got it.

**Comment by kadoshms** [comment]
yes, in case `cidr_blocks` looks like this `cidr_blocks: ['0.0.0.0/0']` it will evaluate to `cidr_blocks: '0.0.0.0/0'` on the graph (while `cidr_blocks: ['0.0.0.0/0', '0.0.0.0/0']` will remain `cidr_blocks: ['0.0.0.0/0', '0.0.0.0/0']`) so it's in-consistent

**Review by rotemavni — APPROVED** [review]
@kadoshms 👏 👏 👏

**Pull request merged**